### PR TITLE
[TRAFFIC-9320] [TRAFFIC-9321] - Add `confluent network peering update|delete` command

### PR DESCRIPTION
**Pull request by syoung-cflt**
What
----

Note that this is merging into the feature branch https://github.com/confluentinc/cli/tree/traffic-8850-network-cli. We will merge commits into main from this feature branch after all of the commands for network are implemented
* Add `confluent network peering update` to update the display name of a peering connection.
* Add `confluent network peering delete` to delete a peering connection.

References
----------

[CLI Commands Design](https://confluentinc.atlassian.net/wiki/spaces/PM/pages/2987394300/1-pager+--+Network+CLI)
[Implementation 1-Pager](https://confluentinc.atlassian.net/wiki/spaces/~712020463029eabbac401bbe1b033ce0d3a00e/pages/3136749631/Network+CLI)

Test & Review
-------------

Added integration tests.

Performed manual tests

Open Questions / Follow-ups
---------------------------

### CODE REVIEW

File: internal/network/command_peering_update.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		RunE:              c.update,
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: `Update the name of peering "peer-111111"`,
+				Code: `confluent network peering update peer-111111 --name "new name"`,
+			},
+		),
+	}
+
+	cmd.Flags().String("name", "", "Name of the peering.")
```
Comment (brianstrauch) [inline internal/network/command_peering_update.go:27]:
```suggestion
	cmd.Flags().String("name", "", "Name of the peering connection.")
```

File: pkg/resource/resource.go [hunk 2]
```diff
@@ -35,6 +35,7 @@ const (
 	MirrorTopic                 = "mirror topic"
 	Network                     = "network"
 	Organization                = "organization"
+	Peering                     = "peering"
```
Comment (brianstrauch) [inline pkg/resource/resource.go:38]: Do we want to call this a "peering connection"?

Comment (syoung-cflt) [reply pkg/resource/resource.go:38]: Looks like our api documentation uses just "peering", should I change all of the "peering connection" to "peering" if thats the case, to keep things consistent?

Comment (brianstrauch) [reply pkg/resource/resource.go:38]: yes please!

Comment (syoung-cflt) [reply pkg/resource/resource.go:38]: Thank you! Removed the word `connection` and updated the relevant golden files as well.

**Review by ejagade — APPROVED** [review]

**Review by brianstrauch — APPROVED** [review]

**Pull request merged**